processor/batchprocessor/batch_processor.go

```diff
@@ -182,15 +182,11 @@ func (bp *batchProcessor) Capabilities() consumer.Capabilities {
 
 // Start is invoked during service startup.
 func (bp *batchProcessor) Start(context.Context, component.Host) error {
-	bp.goroutines.Add(1)
```

Comment:
No go-routine started, no need to Add then Done in Shutdown, which will cause issues if Shutdown is called without Start for example.